astropy/io/fits/file.py

```diff
@@ -436,10 +440,14 @@ def _open_filename(self, filename, mode, clobber):
         elif ext == '.zip' or magic.startswith(PKZIP_MAGIC):
             # Handle zip files
             self._open_zipfile(self.name, mode)
+        elif ext == '.bz2' or magic.startswith(BZIP2_MAGIC):
+            # Handle bzip2 files
+            self._file = bz2.BZ2File(self.name, PYFITS_MODES[mode])
+            self.compression = 'bzip2'
```

Comment:
This `self.compression` attribute was already set on line 151 and doesn't need to be set again.